src/App.tsx

```diff
@@ -1,25 +1,43 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { BrowserRouter as Router, Routes, Route, Navigate } from 'react-router-dom';
+import Cookies from 'js-cookie';
+import ProtectedRoute from 'components/ProtectedRoute';
+import Login from 'pages/Login/Login';
+import Signup from 'pages/Signup/Signup';
+import Dashboard from 'pages/Dashboard';
+import Student from 'pages/Student';
+
+const isAuthenticated = () => {
+  return !!Cookies.get('token');
+};
 
 function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <Routes>
+        <Route
+          path="/"
+          element={
+            isAuthenticated() ? <Navigate to="/dashboard" replace /> : <Login />
+          }
+        />
+        <Route
+          path="/signup"
+          element={
+            isAuthenticated() ? <Navigate to="/dashboard" replace /> : <Signup />
+          }
+        />
+        <Route
+          path="/login"
+          element={
+            isAuthenticated() ? <Navigate to="/dashboard" replace /> : <Login />
```

Comment:
can we make this component clean and handles all routing in a separate component?